src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -401,13 +401,10 @@ private LazyItemList GetItemList(string itemType)
             return ret;
         }
 
-        public IList<ItemData> GetAllItems()
+        public IEnumerable<ItemData> GetAllItems()
```

Comment:
Might be worth at least commenting that this is deferred, so that callers can be careful about multiple enumerations.  Or consider changing the name to make it more clear.

Reply:
I'll change the name.